uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/menu/MenuBuilder.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.client.docks.view.menu;
+
+import com.github.gwtbootstrap.client.ui.Button;
+import com.github.gwtbootstrap.client.ui.Dropdown;
+import com.github.gwtbootstrap.client.ui.DropdownButton;
+import com.github.gwtbootstrap.client.ui.NavLink;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import org.jboss.errai.security.shared.api.identity.User;
+import org.uberfire.security.authz.AuthorizationManager;
+import org.uberfire.workbench.model.menu.*;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.gwtbootstrap.client.ui.resources.ButtonSize.MINI;
+
+@ApplicationScoped
+public class MenuBuilder {
+
+    @Inject
+    private AuthorizationManager authzManager;
+
+    @Inject
+    private User identity;
+
+
+    public Widget makeItem(final MenuItem item,
+                           boolean isRoot) {
```

Comment:
Personally I would split this method to shorter methods. I'll merge anyway, not a cause for rejection.